install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,476 @@ toc::[]
+when the variable is missing or *nil*. When `ROUTER_MAX_CONNECTIONS` is not
+set or is *nil*, 20000 is used. Environment variables are a map where the key
+is the environment variable name and the content is the value of the string.
+
+See .xref:../../architecture/core_concepts/routes.adoc#env-variables[Route-specific Environment variables]
+for more information.
+
+
+[[example]]
+
+=== Pulling It All Together, An Example
+
+Here is a simplistic template based on the HAProxy template file.
+
+First off lets start with a comment:
+
+----
+{{/*
+  Here is a small example of how to work with templates
+  taken from the haproxy template file
+*/}}
+----
+
+Lets start to create an output file. The define construct gives the
+file name and everything up to the closing "end" contributes to the file.
+
+----
+{{ define "/var/lib/haproxy/conf/haproxy.config" }}
+global
+{{ end }}
+----
+
+The above will copy *global* to the */var/lib/haproxy/conf/haproxy.config* file
+and then close the file.
+
+Lets set up logging based on environment variables.
+
+----
+{{ with (env "ROUTER_SYSLOG_ADDRESS" "") }}
+  log {{.}} {{env "ROUTER_LOG_FACILITY" "local1"}} {{env "ROUTER_LOG_LEVEL" "warning"}}
+{{ end }}
+----
+
+The environment variables are a map with the environment variable name as
+the key and the contents as the value. The *env* function extracts the value
+for the environment variable.  if the environment variable is not defined or
+*nil* the second argument is returned.
+
+The the result of evaluating what is between the "()" goes in "." (dot).
+The *with* action tests Dot for *nil*. If not *nil* the clause
+is processed up to the *end*. In the above assume `ROUTER_SYSLOG_ADDRESS`
+contains "/var/log/msg", `ROUTER_LOG_FACILITY` is not defined and
+`ROUTER_LOG_LEVEL` contains "info". The following will be copied to the
+output file:
+
+----
+  log /var/log/msg local1 info
+----
+
+Each admitted route ends up generating lines in the configuration file.
+Lets go through the admitted routes:
+
+----
+{{ range $cfgIdx, $cfg := .State }}
+  backend be_http_{{$cfgIdx}}
+{{end}}
+----
+
+*.State* is a map of *ServiceAliasConfig* the key is the route name.
+*range* steps through the map and for each pass it sets $cfgIdx with the *key*,
+and sets $cfg to point to the *ServiceAliasConfig* that describes the route. If
+there are two routes named *myroute*, *hisroute*, the above will copy
+
+----
+  backend be_http_myroute
+  backend be_http_hisroute
+----
+
+to the output file.
+
+Route Annotations, *$cfg.Annotations*, is also a map with the annotation
+name as the key and the content string as the value.  The route can have
+as many annotations as desired and the use is defined by the developer.
+The user codes the annotation into the route and the developer customized
+the HAProxy template to handle the annotation.
+
+The common usage is to index the annotation to get the value. (*env* does
```

Comment:
How about, "The common usage is to index the annotations to get the value, similar to how we use *env* to get the value of an environment variable."? Or maybe drop the simile because the syntax is a bit different.